### PR TITLE
Add post on setting up Docker in FreeBSD

### PR DESCRIPTION
**Pull request by BonfaceKilz**

### CODE REVIEW

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+draft = false
++++
+
+Right now, you can't run docker natively on FreeBSD. I understand that there are arguably better alternatives to Docker in the FreeBSD universe like using jails, but it's hard to adjust to this if you're coming from Linux. Also, for many projects out here, Docker is the preferred containerisation method of use, so rest assured, alot of READMEs out here have instructions for a Docker set-up, not jails.
+
+Some people would ask, why not build Docker from the source. Well, that's not possible for now because Docker uses some Linux Kernel features like namespaces and cgroups which are absent in FreeBSD. To make this work natively in FreeBSD, someone would have to work around this Linux kernel coupling. This has been attempted [before](https://www.freshports.org/sysutils/docker-freebsd/) but the project has been stale for some time now. The only practical solution left is running Docker inside a Linux VM inside FreeBSD. In this post, we'll be using `docker-machine` and `virtualbox` to do just that.
+
+First, we need to install a docker-client, docker-machine and virtualbox. To install docker(this installs the docker-client), docker-machine and virtualbox:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: The two sentences in this are redundant. I'd rephrase this as:

```
First, we need to install a docker-client, docker-machine and virtualbox by:
```

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+```bash
+sudo pkg install docker docker-machine virtualbox-ose
+```
+
+After installing virtualbox, you need to load the `vboxdrv` kernel module by adding `vboxdrv_load="YES"` in `/boot/loader.conf`. You should also add your current users to your `vboxusers` group in order to use vbox:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Replace `users` with `user`

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+```nil
+sudo pw groupmod vboxuser -m <username>
+```
+
+Once you are done, reboot your workstation in order to load the virtualbox kernel modules.
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: You can continue with the previous statement instead of using `Once you are done`, for example:

```
and then reboot your workstation...
```

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Docker Machine lets you easily create Docker hosts on your computer. It creates servers, installs Docker on them, then configures the Docker client to talk to them[0]. I decided to use docker-machine for the docker set-up because it's painless to use and really convenient. To create a virtualbox host locally using Docker Machine:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: The first three sentences here speak of the advantages of docker-machine. So I think it would be better if you start with:

```
I decided to use docker-machine for setting up docker because it lets me easily create Docker hosts on my computer by creating servers, installing docker on then and configuring Docker client to talk to them [0] and it's painless to use
```

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+```bash
+# Create the virtual box host
+docker-machine create -d virtualbox default
+
+# List all the hosts present:
+docker-machine ls
+```
+
+Now you should set up your environment variables, so that your docker client can be able to use docker installed inside virtualbox:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Change of POV. You should just stick with 'I'

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+```bash
+eval "$(docker-machine env default)"
+```
+
+You can opt to stick that fella ^ in your `.bashrc` or `.zshrc` file. With that, you can run docker in FreeBSD :)
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Change of POV

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+description = "A guide on how to run Docker in FreeBSD"
+date = 2020-01-19T19:09:00+03:00
+tags = ["freebsd", "docker", "howto"]
+draft = false
++++
+
+Right now, you can't run docker natively on FreeBSD. I understand that there are arguably better alternatives to Docker in the FreeBSD universe like using jails, but it's hard to adjust to this if you're coming from Linux. Also, for many projects out here, Docker is the preferred containerisation method of use, so rest assured, alot of READMEs out here have instructions for a Docker set-up, not jails.
+
+Some people would ask, why not build Docker from the source. Well, that's not possible for now because Docker uses some Linux Kernel features like namespaces and cgroups which are absent in FreeBSD. To make this work natively in FreeBSD, someone would have to work around this Linux kernel coupling. This has been attempted [before](https://www.freshports.org/sysutils/docker-freebsd/) but the project has been stale for some time now. The only practical solution left is running Docker inside a Linux VM inside FreeBSD. In this post, we'll be using `docker-machine` and `virtualbox` to do just that.
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]:
```
Some would ask, why not build Docker from source?
```
has a better ring to it

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
++++
+title = "How to setup docker in FreeBSD"
+description = "A guide on how to run Docker in FreeBSD"
+date = 2020-01-19T19:09:00+03:00
+tags = ["freebsd", "docker", "howto"]
+draft = false
++++
+
+Right now, you can't run docker natively on FreeBSD. I understand that there are arguably better alternatives to Docker in the FreeBSD universe like using jails, but it's hard to adjust to this if you're coming from Linux. Also, for many projects out here, Docker is the preferred containerisation method of use, so rest assured, alot of READMEs out here have instructions for a Docker set-up, not jails.
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Your lines are really long. I'm of the opinion that you need to have shorter lines e.g. wrapping at 80 chars. You can see a discussion of the same here: https://about.gitlab.com/blog/2016/10/11/wrapping-text/

**Review by jnduli — CHANGES_REQUESTED** [review]

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md (hunk 3 above)
Comment (BonfaceKilz) [reply content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: This is the beginning of a completely new sentence after a src block so I'm going to leave it as is.

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md (hunk 5 above)
Comment (BonfaceKilz) [reply content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: It's not really a change of POV. In the previous paragraph, I used "I" to show why I prefer docker-machine. In other paragraphs, it reads like a tutorial.

**Comment by BonfaceKilz** [comment]
@jnduli done. I've incorporated most of the changes. I've separated the 80-chars wrap into a commit of its own to make reviewing a bit easier.

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+arguably better alternatives to Docker in the FreeBSD universe like using jails,
+but it's hard to adjust to this if you're coming from Linux. Also, for many
+projects out here, Docker is the preferred containerisation method of use, so
+rest assured, alot of READMEs out here have instructions for a Docker set-up,
+not jails.
+
+Some people would ask, why not build Docker from source? Well, that's not
+possible for now because Docker uses some Linux Kernel features like namespaces
+and cgroups which are absent in FreeBSD. To make this work natively in FreeBSD,
+someone would have to work around this Linux kernel coupling. This has been
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Remove `this Linux Kernel coupling` and just use `this` to avoid repetition.

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+attempted [before](https://www.freshports.org/sysutils/docker-freebsd/) but the project has been stale for some time now. The only
+practical solution left is running Docker inside a Linux VM inside FreeBSD. In
+this post, we'll be using `docker-machine` and `virtualbox` to do just that.
+
+First, we need to install a docker-client, docker-machine and virtualbox by:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md]: Replace we with you so as to remain consistent with the tutorial feel.

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+```bash
+sudo pw groupmod vboxuser -m <username>
+```
+
+Once you are done, reboot your workstation in order to load the virtualbox
+kernel modules.
+
+I decided to use docker-machine for setting up docker because it lets me easily
+create Docker hosts on my computer by creating servers, installing docker on
+them and configuring Docker client to talk to them [0] and it's painless to use:
```
Comment (jnduli) [inline content/post/2020-01-20-how-to-setup-docker-in-freebsd.md:43]: Perhaps you can make [0] a link to the bottom page. Something liek an anchor https://gist.github.com/asabaylus/3071099

**Review by jnduli — CHANGES_REQUESTED** [review]

File: content/post/2020-01-20-how-to-setup-docker-in-freebsd.md (hunk 11 above)
Comment (BonfaceKilz) [reply content/post/2020-01-20-how-to-setup-docker-in-freebsd.md:43]: This applies for gh-flavored md. Also consider I'm exporting things from an org file to md. I'm going to leave this as is :sweat_smile:

**Review by jnduli — APPROVED** [review]
LGTM
